src/White.jsx

```diff
@@ -0,0 +1,174 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCircle } from "@fortawesome/free-solid-svg-icons";
+
+export default function White(props) {
+  const { AllshoesName, AllImage } = props;
+
+  return (
+    <>
+      <div className="container">
+        <div className="listContainer">
+          {AllshoesName.map((ele) => {
+            if (ele.color === "WHITE") {
+              return (
+                <div className="infoCard">
+                  <img alt="shoes" width="100px" src={AllImage[ele.id]} />
+                  <p className="name">{ele.name}</p>
+                  <p className="color">
+                    {ele.color}{" "}
+                    <FontAwesomeIcon
+                      icon={faCircle}
+                      size="lg"
+                      color={ele.color}
+                    />
+                  </p>
+                  <p className="type">{ele.type}</p>
+                  <p className="year">{ele.where_to_bought}</p>
+                  <p className="where">{ele.year_bounght}</p>
+                </div>
+              );
+            } else {
+              return null;
+            }
+          })}
+        </div>
+      </div>
+      <div className="container">
+        <div className="listContainer">
+          {AllshoesName.map((ele) => {
+            if (ele.color === "RED") {
```

Comment:
What I think you wanted to do, is show each color one at a time. You can use `Array.sort()` for this! For example:

```jsx
const shoesArray = Object.values(AllshoesName); // get all the shoes as an array
shoesArray.sort((a, b) => a.color < b.color ? 1 : -1); // sort them alphabetically by color

return (
  <div>
  {shoesArray.map(shoe => <InfoCard key={shoe.id} shoe={shoe} />)}
  </div>;
);
```

Reply:
Here is another way: you can decide which order you want the colors to be in, and then use `filter` and `map` to make elements just for those colors:

```jsx
const COLOR_ORDER = [
  "WHITE",
  "BLACK",
  "GREEN",
  "RED",
  "BLUE",
];
const shoes = Object.values(AllshoesName);

return (
  <div>
    {COLOR_ORDER.map(color => (
       <div key={color}>
         {shoes.filter(shoe => shoe.color === color).map(shoe => (
           <InfoCard key={shoe.id} shoe={shoe} />
         )}
       </div>
    )}
  </div>
);
```